File: Probas de lenguajes/Ejemplo.php
<?php
declare(strict_types=1); // Fuerza el tipado estricto

function suma(int $a, int $b) {
    return $a + $b;
}

$a = 2;
$b = "3"; // Esto lanzará un TypeError al pasarlo a la función
echo "Voy a sumar $a y $b\n";
echo "¿debería ser 5?\n";

echo suma($a, $b) . "\n";
?>

File: Probas de lenguajes/Ejemplo.java
public class Ejemplo {
  public static int suma(int a, int b) {
    return a + b;
  }

  public static void main(String[] args) {
    int a = 2;
    int b = 3;  //String b = "3";

    System.out.printf("Voy a sumar %d y %d%n", a, b);   //System.out.printf("Voy a sumar %d y %s%n", a, b);
    System.out.println("¿debería ser 5?");

    System.out.println(suma(a, b));
  }
}